src/jit/flowgraph.cpp

```diff
@@ -9021,10 +9036,76 @@ void Compiler::fgSimpleLowering()
                     // Add in a call to an error routine.
                     fgSetRngChkTarget(tree, false);
                 }
+
+#if FEATURE_FIXED_OUT_ARGS
+                if (tree->gtOper == GT_CALL)
+                {
+                    GenTreeCall* call = tree->AsCall();
+                    // Fast tail calls use the caller-supplied scratch
+                    // space so have no impact on this method's outgoing arg size.
+                    if (!call->IsFastTailCall())
+                    {
+                        // Update outgoing arg size to handle this call
+                        const unsigned thisCallOutAreaSize = call->fgArgInfo->GetOutArgSize();
+                        assert(thisCallOutAreaSize >= MIN_ARG_AREA_FOR_CALL);
+
+                        if (thisCallOutAreaSize > outgoingArgSpaceSize)
+                        {
+                            outgoingArgSpaceSize = thisCallOutAreaSize;
+
+                            // If a function has localloc, we will need to move the outgoing arg space when the
+                            // localloc happens. When we do this, we need to maintain stack alignment. To avoid
+                            // leaving alignment-related holes when doing this move, make sure the outgoing
+                            // argument space size is a multiple of the stack alignment by aligning up to the next
+                            // stack alignment boundary.
+                            if (compLocallocUsed)
+                            {
+                                outgoingArgSpaceSize = (unsigned)roundUp(outgoingArgSpaceSize, STACK_ALIGN);
+                            }
+
```

Comment:
You could pull the localloc update/alignment down below, just before setting the actual size.